### PR TITLE
feat: add getLogger to app and ctx

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

app and ctx

##### Description of change

it can be override with custom function

Closes eggjs/egg#36

### CODE REVIEW

**Comment by popomore** [comment]
看来 eslint break 了 https://github.com/eslint/eslint/pull/7403

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
rebase 一下

**Comment by popomore** [comment]
把 circle ci 关了？

**Comment by fengmk2** [comment]
circle 关了

File: test/utils.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ exports.startLocalServer = () => {
     });
   });
 };
+process.once('exit', () => localServer.close());
```
Comment (fengmk2) [inline test/utils.js]: 好吧，mocha 好像不管是否还有 server，都是自己退出的。

File: test/lib/core/urllib.test.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const urllib = require('../../../lib/core/urllib');
 
 describe('test/lib/core/urllib.test.js', () => {
   let client;
-  const url = 'https://a.alipayobjects.com/aliBridge/1.0.0/aliBridge.min.js';
+  const url = 'http://www.google.com/generate_204';
```
Comment (fengmk2) [inline test/lib/core/urllib.test.js]: 改成 local 的吧，要不然本地运行这个 ci google 别墙，跑不过。

**Comment by popomore** [comment]
改了

**Comment by fengmk2** [comment]
+1

**Pull request merged**